Autoconfigure docs (#5436)
sonata-project/SonataAdminBundle: docs/cookbook/recipe_auto_configure_admin_classes.rst

Auto Configuring Admin Classes
==============================

If you have a lot of admin classes and don't want to write all the service
definitions or if maybe you just don't like writing configuration and want

> everything setup for you, this bundle will allow you that.

Reviewer: ```suggestion
everything set up for you, this bundle will allow you not to do that.
```

Download the SonataAutoConfigureBundle
--------------------------------------

.. code-block:: bash

    composer require kunicmarko/sonata-auto-configure-bundle

How to use
----------

The only thing you need to do is add the configuration for this bundle:

.. code-block:: yaml

    # config/packages/sonata_auto_configure.yaml
    sonata_auto_configure:
        admin:
            suffix: Admin
            manager_type: orm
        entity:
            namespaces:
                - { namespace: App\Entity, manager_type: orm }
        controller:
            suffix: Controller
            namespaces:
                - App\Controller\Admin

.. note::

    Be sure that the admin directory is included in
    auto discovery and that autoconfigure is enabled.

This configuration basically means, find all admin classes,

> remove the ``Admin`` suffix and try to find entity with the

Reviewer: ```suggestion
remove the ``Admin`` suffix and try to find an entity with the
```

> same name in ``App\Entity`` namespace and add the ``orm`` manager_type.

Reviewer: ```suggestion
same name in ``App\Entity`` namespace and add the ``orm`` ``manager_type``.
```

After, try to find the controllers in ``App\Controller\Admin`` but
replace the ``Admin`` suffix for ``Controller``.

> If you are interested in more details how does this bundle work, read it `here`_.

Reviewer: ```suggestion
If you are interested in more details on how this bundle works, read it `here`_.
```

> Annotation

Reviewer: ```suggestion
Annotations
```

> ----------

Reviewer: ```suggestion
-----------
```

> Annotation has a higher priority than bundle guesses, so if you are not happy

Reviewer: ```suggestion
Annotations have a higher priority than bundle guesses, so if you are not happy
```

with something just add the annotation to your admin class::

>     namespace App\Entity;

Reviewer 2: This looks wrong.

Why would you put an `*Admin` class under the `Entity` namespace.

Author: because it was a copy/paste :smile:

>     use KunicMarko\SonataAutoConfigureBundle\Annotation as Sonata;

Reviewer 2: Please sort this
